File: app/src/main/java/org/witness/proofmode/onboarding/WelcomeFragment.java
package org.witness.proofmode.onboarding;

import android.content.Intent;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import org.witness.proofmode.DataLegendActivity;
import org.witness.proofmode.R;


public class WelcomeFragment extends Fragment {
    private View mRootView;

    public static WelcomeFragment newInstance() {
        WelcomeFragment f = new WelcomeFragment();
        Bundle args = new Bundle();
        f.setArguments(args);
        return f;
    }

    public WelcomeFragment() {
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        mRootView = inflater.inflate(R.layout.onboarding_welcome, container, false);

        View btnLearnMore = mRootView.findViewById(R.id.btnLearnMore);
        btnLearnMore.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getActivity(), DataLegendActivity.class);
                getActivity().startActivity(intent);
            }
        });

        View btnNext = mRootView.findViewById(R.id.btnNext);
        btnNext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (getActivity() instanceof OnboardingStepListener) {
                    ((OnboardingStepListener) getActivity()).onNextPressed();
                }
            }
        });

        return mRootView;
    }
}

File: app/src/main/java/org/witness/proofmode/onboarding/WelcomeFragment.kt
package org.witness.proofmode.onboarding

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import org.witness.proofmode.org.witness.proofmode.ui.DataLegendActivity
import org.witness.proofmode.databinding.OnboardingWelcomeBinding

class WelcomeFragment : Fragment() {
    private var _binding: OnboardingWelcomeBinding? = null
    private val binding get() = _binding!!
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = OnboardingWelcomeBinding.inflate(inflater, container, false)
        binding.btnLearnMore.setOnClickListener {
            val intent = Intent(activity, DataLegendActivity::class.java)
            requireActivity().startActivity(intent)
        }
        binding.btnNext.setOnClickListener {
            if (activity is OnboardingStepListener) {
                (activity as OnboardingStepListener).onNextPressed()
            }
        }
        return binding.root
    }

    companion object {
        fun newInstance(): WelcomeFragment {
            val f = WelcomeFragment()
            val args = Bundle()
            f.arguments = args
            return f
        }
    }
}